lib/block-supports/layout.php

```diff
@@ -412,16 +428,16 @@ function gutenberg_render_layout_support_flag( $block_content, $block ) {
 		}
 	}
 
-	/*
-	 * This assumes the hook only applies to blocks with a single wrapper.
-	 * A limitation of this hook is that nested inner blocks wrappers are not yet supported.
-	 */
-	$content = preg_replace(
-		'/' . preg_quote( 'class="', '/' ) . '/',
-		'class="' . esc_attr( implode( ' ', $class_names ) ) . ' ',
-		$block_content,
-		1
-	);
+	$inner_content_classnames = isset( $block['innerContent'][0] ) && 'string' === gettype( $block['innerContent'][0] ) ? gutenberg_get_classnames_from_last_tag( $block['innerContent'][0] ) : '';
```

Comment:
Is it worth adding a comment to explain what the `0` index of `innerContent` is, since it's sort of internal to the structure of the block object? E.g. something along the lines of:

```
// Attempt to extract classnames from last tag in block markup directly preceding first inner blocks marker.
// This identifies the first, deepest nested inner blocks wrapper.
```

Please correct me, though, if I've misunderstood how this is working, I just [looked up the block parser class](https://github.com/WordPress/wordpress-develop/blob/3b63a75108b7f6d5e9112c556285ab1bc80ddbcf/src/wp-includes/class-wp-block-parser.php#L62), so I _think_ that's what it's doing, but could very well be wrong 🙂

From the comment in that class, it sounds like there's the potential for blocks to have another string between `null` markers (on [this line](https://github.com/WordPress/wordpress-develop/blob/3b63a75108b7f6d5e9112c556285ab1bc80ddbcf/src/wp-includes/class-wp-block-parser.php#L62) of the comments). I've only ever encountered an opening string, multiple `null` values for each of the inner blocks, and then a closing string, so I wonder if there are any blocks that do that in practice? If so, I reckon we could look at that if and when we encounter it and then come back to the logic in this function to re-examining it.

For now, looking at the last tag in the opening string sounds good to me!

Reply:
Yes, that's exactly how it works! Good idea to add a comment, as it's not immediately obvious.